auth/user_mgt.go

```diff
@@ -0,0 +1,655 @@
+func (it *UserIterator) PageInfo() *iterator.PageInfo { return it.pageInfo }
+
+// Next returns the next result. Its second return value is [iterator.Done] if
+// there are no more results. Once Next returns [iterator.Done], all subsequent
+// calls will return [iterator.Done].
+func (it *UserIterator) Next() (*ExportedUserRecord, error) {
+	if err := it.nextFunc(); err != nil {
+		return nil, err
+	}
+	user := it.users[0]
+	it.users = it.users[1:]
+	return user, nil
+}
+
+// SetCustomUserClaims sets additional claims on an existing user account.
+//
+// Custom claims set via this function can be used to define user roles and privilege levels.
+// These claims propagate to all the devices where the user is already signed in (after token
+// expiration or when token refresh is forced), and next time the user signs in. The claims
+// can be accessed via the user's ID token JWT. If a reserved OIDC claim is specified (sub, iat,
+// iss, etc), an error is thrown. Claims payload must also not be larger then 1000 characters
+// when serialized into a JSON string.
+func (c *Client) SetCustomUserClaims(ctx context.Context, uid string, customClaims map[string]interface{}) error {
+	if customClaims == nil || len(customClaims) == 0 {
+		customClaims = map[string]interface{}{}
+	}
+	_, err := c.UpdateUser(ctx, uid, (&UserToUpdate{}).CustomClaims(customClaims))
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+// ------------------------------------------------------------
+// Setters and utilities for Create and Update input structs.
+
+func (p *commonParams) appendErrString(field, s string, i ...interface{}) {
+	if p.errors == nil {
+		p.errors = map[string]string{}
+	}
+	if _, ok := p.errors[field]; !ok {
+		p.errors[field] = fmt.Sprintf(s, i...)
+	}
+}
+
+func (p *commonParams) getErrors() error {
+	if p.errors == nil || len(p.errors) == 0 {
+		return nil
+	}
+	var errs []string
+	for _, v := range p.errors {
+		errs = append(errs, v)
+	}
+	return fmt.Errorf(strings.Join(errs, ", "))
+}
+
+func (p *commonParams) payloadInitialized() {
+	if p.payload == nil {
+		p.payload = make(map[string]interface{})
+	}
+}
+
+func (p *commonParams) set(key string, value interface{}) {
+	p.payloadInitialized()
+	p.payload[key] = value
+}
+
+// assumes that payloadName is a string field in p.payload
+func processDeletion(p *commonParams, payloadName, errorName string) error {
+	if dn, ok := p.payload[payloadName]; ok && len(dn.(string)) == 0 {
+		p.addToListParam(deletionSpecs[payloadName].deleteListName, deletionSpecs[payloadName].deleteFieldName)
+		delete(p.payload, payloadName)
+	}
+	return nil
+}
+
+func processClaims(p *commonParams, payloadName, errorName string) error {
+	if _, ok := p.payload[payloadName]; !ok {
+		return nil
+	}
+	if err := p.checkReservedClaims(); err != nil {
+		return err
+	}
+	return p.setCustomAttributes()
+}
+
+func (p *commonParams) checkReservedClaims() error {
+	cc := p.payload["customClaims"]
+	switch claims := cc.(type) {
```

Comment:
This is a really awkward use of a switch case. Simply do a type assertion into a map, and catch any type errors.